From 01eb36585fdb5a12c2870a8c2735d7466ed6f293 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 21 Nov 2019 10:56:58 +0800
Subject: [PATCH 120/126] Input: goodix - Add support for Goodix GT917S

Goodix GT917S is a touchscreen chip from Goodix that is in the GT1x
family.

Add its support by assigning the gt1x config to it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/input/touchscreen/goodix.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index 8b06780b37e5..abfa00d546b3 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -230,7 +230,8 @@ static const struct goodix_chip_data *goodix_get_chip_data(const char *id)
 {
 	if (!strcmp(id, "1151") ||
 	    !strcmp(id, "5663") ||
-	    !strcmp(id, "5688"))
+	    !strcmp(id, "5688") ||
+	    !strcmp(id, "917S"))
 		return &gt1x_chip_data;
 
 	if (!strcmp(id, "911") ||
@@ -1038,6 +1039,7 @@ static const struct of_device_id goodix_of_match[] = {
 	{ .compatible = "goodix,gt911" },
 	{ .compatible = "goodix,gt9110" },
 	{ .compatible = "goodix,gt912" },
+	{ .compatible = "goodix,gt917s" },
 	{ .compatible = "goodix,gt927" },
 	{ .compatible = "goodix,gt9271" },
 	{ .compatible = "goodix,gt928" },
-- 
2.23.0

